akeneo-design-system/src/components/Pagination/Pagination.tsx

```
@@ -0,0 +1,181 @@
+import React, {FC} from 'react';
+import styled, {css} from 'styled-components';
+import {getColor, AkeneoThemedProps} from '../../theme';
+
+type PaginationProps = {
+  /**
+   * The current page number (starting à 1).
+   */
+  currentPage: number;
+
+  /**
+   * The total number of pages.
+   */
+  itemsTotal: number;
+
+  /**
+   * The number of items per page. Default to 25.
+   */
+  itemsPerPage?: number;
+
+  /**
+   * Handler called when a pagination item is clicked.
+   */
+  onClick: (page: number) => void;
+};
+
+const MAX_PAGINATION_ITEMS_WITHOUT_SEPARATOR = 4;
+const PAGINATION_SEPARATOR = '...';
+
+const Pagination: FC<PaginationProps> = ({currentPage, itemsTotal, itemsPerPage = 25, onClick}) => {
+  if (itemsPerPage <= 0) {
```

I think that in this case it's more a developper error and should throw an exception.
IMHO, having no item is something that can happen (if there is no content) but having pages of zero items is an error coming from the developer. What do you think?